cub/util_device.cuh

```diff
@@ -348,6 +366,19 @@ CUB_RUNTIME_FUNCTION inline cudaError_t PtxVersionUncached(int& ptx_version)
 
         result = cudaFuncGetAttributes(&empty_kernel_attrs,
                                        reinterpret_cast<void*>(empty_kernel));
+        // Any failure means either no GPU is available or the provided device
+        // binary does not match the generated function code
+        if (result != cudaSuccess) {
+            int sm_version;
+            int device;
+            if (!CubDebug(cudaGetDevice(&device))) {
+                if (!CubDebug(SmVersionUncached(sm_version, device))) {
+                    printf("Incompatible GPU: you are trying to run this program on sm_%d%d, "
+                            "different from the one that it was compiled for\n",
+                            sm_version/100, (sm_version%100)/10);
```

Comment:
`printf` isn't a very robust or canonical way of reporting errors. This should really be an exception, but CUB doesn't currently throw exceptions.